app/main/views/users.py

```diff
@@ -271,71 +281,123 @@ def export_users_for_framework(framework_slug):
 
     suppliers_with_a_complete_service = frozenset(framework.get_supplier_ids_for_completed_service())
 
-    supplier_id_published_service_count = dict(db.session.query(
-        Service.supplier_id,
-        func.count(Service.id)
+    users = db.session.query(
+        User.supplier_id, User.name, User.email_address, User.user_research_opted_in,
+        SupplierFramework.framework_id, SupplierFramework.supplier_id,
+    ).filter(
+        SupplierFramework.supplier_id == User.supplier_id
     ).filter(
-        Service.status == 'published',
-        Service.framework_id == framework.id
-    ).group_by(
-        Service.supplier_id
-    ).all())
+        SupplierFramework.framework_id == framework.id
+    ).filter(
+        User.active.is_(True)
+    ).all()
+
+    users_for_each_supplier = {}
+    for u in users:
+        user_obj = {
+            'email_address': u.email_address,
+            'name': u.name,
+            'user_research_opted_in': u.user_research_opted_in,
+        }
+        # If supplier id key already exists, append the user to that supplier's list
+        users_for_each_supplier.setdefault(u.supplier_id, []).append(user_obj)
+
+    lot_slugs_by_id = {lot.id: lot.slug for lot in framework.lots}
+
+    # Creates a dictionary of dictionaries, with published service counts per lot per supplier.
+    # To do this we use dictionary comprehension, SQL query and itertools' 'groupby' function.
+    # Grouping by supplier id is more efficient as this is what we will be iterating over later.
+    service_counts_by_lot_by_supplier = {
```

Comment:
`groupby` is tricky. This will (I think) create an entry per supplier id - lot comination
Have a look at the output of:
```
[i for i in groupby([1,2,2,2,2,3,3,4,4,4,5,5,5,4,4,3,2,2,1,2,2,3,3,4,4,5,1,1,1,1,2,2,3,3,4,4,5,5,3,2,3,3,4,5])]
```
For an example. Is this what we want here?

Reply:
Actually that made me notice that the query used here doesn't have an `order_by` set, and while I think the result of a `group_by` will almost always come out sorted by major group column first, I don't think it by law *has to*, so it would be better to add an `.order_by(Service.supplier_id)` clause here.